lua/nvim_lsp/rust_analyzer.lua

```diff
@@ -1,11 +1,88 @@
 local configs = require 'nvim_lsp/configs'
 local util = require 'nvim_lsp/util'
+local server_name = 'rust_analyzer'
+local release_url = 'https://api.github.com/repos/rust-analyzer/rust-analyzer/releases/latest'
+local postfix = '-linux'
+local bin = 'rust-analyzer'
+
+if vim.fn.has('mac') == 1 then
+  postfix = '-mac'
+elseif vim.fn.has('win32') == 1  or vim.fn.has('win64') == 1 then
+  postfix = '-windows.exe'
+end
+
+local function make_installer()
+  local I = {}
+  local P = util.path.join
+  local install_dir = P{util.base_install_dir, server_name}
+	local cmd_path = P{install_dir, bin}
+
+  local function getReleaseJson()
+    local json_string = vim.fn.system(string.format('curl -s "%s"', release_url), install_dir)
+    return vim.fn.json_decode(json_string)
+  end
+
+  function I.install()
+    local install_info = I.info()
+    vim.fn.mkdir(install_info.install_dir, "p")
+    local json = getReleaseJson()
+
+    local download_url = ''
+    local tag_name = json.tag_name
+    for _, value in pairs(json.assets) do
+      if (value ~= nil and string.match(value.browser_download_url, postfix..'$')) then
+        download_url = value.browser_download_url
+      end
+    end
+    if (download_url == '') then
+      print('Could not find download url. Aborting.')
+      return
+    end
```

Comment:
Not sure why you're bothering with this since the download URL can be computed given the version tag and platform (i.e. mac, linux  etc):

```
https://github.com/rust-analyzer/rust-analyzer/releases/download/$VERSION/rust-analyzer-$PLATFORM
```

Reply:
I changed this in d1bac250. Thanks!